hnn_core/optimization.py

```diff
@@ -447,14 +447,18 @@ def _myoptrun(new_params):
         # tiny negative weights are possible. Clip them to 0.
         opt_results[opt_results < 0] = 0
 
-        # update opt_params for the next round
-        for var_name, value in zip(opt_params['ranges'], opt_results):
-            opt_params['ranges'][var_name]['initial'] = value
+        # update opt_params for the next round if total rmse decreased
+        new_avg_rmse = _rmse(opt_dpls['best_dpl'],
+                             opt_dpls['target_dpl'],
+                             tstop=params['tstop'],
+                             weights=None)
+        if new_avg_rmse <= avg_rmse:
```

Comment:
nitpick, can you call `avg_rmse` to be `best_rmse` and `new_avg_rmse` to be `avg_rmse` ?